redis-hbo-provider/src/main/java/com/facebook/presto/statistic/RedisProviderPlugin.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statistic;
+
+import com.facebook.airlift.bootstrap.Bootstrap;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.Plugin;
+import com.facebook.presto.spi.statistics.EmptyPlanStatisticsProvider;
+import com.facebook.presto.spi.statistics.HistoryBasedPlanStatisticsProvider;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import io.lettuce.core.cluster.RedisClusterClient;
+import org.weakref.jmx.guice.MBeanModule;
+
+import javax.management.MBeanServer;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.NoSuchFileException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.facebook.presto.statistic.PropertiesUtil.loadProperties;
+import static org.weakref.jmx.ObjectNames.generatedNameOf;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
+
+public class RedisProviderPlugin
+        implements Plugin
+{
+    private static final Logger log = Logger.get(RedisProviderPlugin.class);
+    private final Map<String, String> propertyMap;
+    public RedisProviderPlugin()
+    {
+        // This plugin cannot be loaded like other plugins because this is not a connector
+        Map<String, String> propertyMap;
+        try {
+            propertyMap = loadAndProcessProperties();
+        }
+        catch (NoSuchFileException fileNotFoundException) {
+            propertyMap = null;
+            log.info("Redis Provider Plugin was not loaded since config file not found");
+        }
+        this.propertyMap = propertyMap;
+    }
+
+    public Iterable<HistoryBasedPlanStatisticsProvider> getHistoryBasedPlanStatisticsProviders()
+    {
+        if (this.propertyMap == null) {
+            log.info("Redis Provider Plugin returned an empty instance");
+            return ImmutableList.of(EmptyPlanStatisticsProvider.getInstance());
+        }
+        if (!(propertyMap.getOrDefault(RedisProviderConfig.COORDINATOR_PROPERTY_NAME, "false").equals("true") &&
+                propertyMap.getOrDefault(RedisProviderConfig.HBO_PROVIDER_ENABLED_NAME, "false").equals("true"))) {
+            return ImmutableList.of(EmptyPlanStatisticsProvider.getInstance());
+        }
+        Injector injector = intializeAndGetInjector();
```

Comment:
Technically this could be done in constructor but I think doing in getHistoryBasedPlanStatisticsProviders makes more sense since this is where the provider is loaded at runtime